crates/sb_workers/user_workers.js

```diff
@@ -36,11 +36,8 @@ class UserWorker {
     async fetch(req) {
         const { method, url, headers, body } = req;
 
-        const bodyData = (await new Response(body).arrayBuffer());
-        const bodyLength = bodyData.byteLength;
-
         const headersArray = Array.from(headers.entries());
-        const hasBody = bodyLength > 0;
+        const hasBody = headersArray.some(([headerName, headerValue]) => headerName.toLowerCase() === 'content-length' && Number(headerValue) > 0);
```

Comment:
const hasBody = (body !== null) && (!bodyUsed);

Reply:
I will add the `body !== null`. `!(bodyUsed)` is unnecessary, but checking if body is null is okay. Note that for the time being, `body` will always have a value. It's supposed to be null when it's empty, but the tech behind this is not respecting the spec. It's an upstream

Reply:
The reason for checking !bodyUsed is that if the body stream has already been read, you are piping a closed stream to the Rust worker, which I would imagine could trigger all sorts of errors

Reply:
> `body` will always have a value. It's supposed to be null when it's empty, but the tech behind this is not respecting the spec. It's an upstream

Where does the request come from ? Isn't it a JS object coming from Deno's std/http/server ?

Reply:
At the time of using `bodyUsed` it's never been read. In fact we need to figure out whether `hasBody` is true or false before we even pipe it. If we for some reason consume the body and then pipe it, tests and everything will fail. Which is why it's designed -> Figure out whether it has a body _without consuming it_ -> Pipe it.